vendor/k8s.io/klog/v2/klog.go

```diff
@@ -919,18 +945,22 @@ func (l *loggingT) output(s severity, log logr.Logger, buf *buffer, file string,
 				}
 			}
 
-			switch s {
-			case fatalLog:
-				l.file[fatalLog].Write(data)
-				fallthrough
-			case errorLog:
-				l.file[errorLog].Write(data)
-				fallthrough
-			case warningLog:
-				l.file[warningLog].Write(data)
-				fallthrough
-			case infoLog:
-				l.file[infoLog].Write(data)
+			if l.oneOutput {
+				l.file[s].Write(data)
```

Comment:
is it possible for s to be something other than fatalLog, errorLog, warningLog or infoLog? previously, values of `s` other than those were ignored

Reply:
No, severity is internal alias for int32. Users of klog cannot pass it's value, it's inferred from function call `klog.Info` calls subfunction with severity `infoLog`.

This code changes is meant to change fallthrough logic. Previously we wrote to files with severtity equal lower to one passed. `one-output` is meant to change this behavior and allow writing to only files matching severity. 
https://github.com/kubernetes/klog/pull/141